aws/request/request.go

```diff
@@ -77,13 +75,10 @@ func New(cfg aws.Config, clientInfo metadata.ClientInfo, handlers Handlers,
 		p = "/"
 	}
 
-	httpReq, _ := http.NewRequest(method, "", nil)
-
-	var err error
-	httpReq.URL, err = url.Parse(clientInfo.Endpoint + p)
+	httpReq, err := http.NewRequest(method, clientInfo.Endpoint+p, nil)
 	if err != nil {
-		httpReq.URL = &url.URL{}
 		err = awserr.New("InvalidEndpointURL", "invalid endpoint uri", err)
+		return &Request{Error: err}
```

Comment:
My take on this is the following:
- the function itself should have been returning a `*Request, error` pair from the beginning. It doesn't do so so I'd rather not break it yet.
- regarding the issue in #689, I don't see why this would make any difference to be honest. In both cases if there's an error there, the API call can't continue. If the developer can't provide a proper URL then it's ok for this to crash. We are also pretty close to Go 1.7 so there should be little to no incentive to support Go 1.4 (or really anything under 1.6 at this stage).